mage_ai/frontend/components/shared/Table/index.tsx

```diff
@@ -428,11 +428,13 @@ function Table({
           noHover={!(linkProps || onClickRow || renderExpandedRowWithObject)}
           // @ts-ignore
           onClick={(e) => {
-            if (onClickRow) {
-              handleRowClick(rowIndex, e);
-            }
+            if (!isInteractiveElement(e)) {
```

Comment:
This is a broader change to the shared table to not trigger the `onClickRow` functionality if the clicked element is an interactive element (e.g. a button in the row). It should capture most (but not all) cases of interactive elements. 

We also don't have to do this change and can individually stop event propagation in the individual interactive element's event handlers as an alternative. Let me know what you think works best!

Reply:
Including this sounds great. In the use case of our Pipeline Runs table, does adding this mean that the row should not be selected (and highlighted with the black background) when clicking the Retry button under the `Status` column even if we remove the `e.stopPropagation()` line of code? Or do we also have to pass in the `event` for the row cells or something?

Reply:
Oh this change shouldn't affect the current behavior of clicking the Retry button! The main idea was if you click on any interactive element in the row, it wouldn't also highlight the row. Clicking on the button under the Status column (e.g. Ready, Done) pops up the retry menu and doesn't highlight the row. However, clicking on a button within the retry menu should still have the same behavior! Only the top-level element within the row matters. Let me know if I can clarify more!